fmn/consumer/backends/sse.py

```diff
@@ -0,0 +1,242 @@
+from __future__ import absolute_import, unicode_literals
+
+from gettext import gettext as _
+import json
+import logging
+import uuid
+import datetime
+
+import pika
+import pytz
+import six
+import fedmsg.meta
+from pika.adapters import twisted_connection
+from twisted.internet import reactor, protocol, defer
+
+from fmn.consumer.backends.base import BaseBackend, shorten
+
+
+_log = logging.getLogger(__name__)
+
+
+class SSEBackend(BaseBackend):
+    """
+    This backend is responsible for the creation of RabbitMQ exchanges and
+    queues and the re-publication of messages for consumption by the Server-
+    Sent Events server.
+
+    Messages are pulled off the "backend" queue and passed to this class, which
+    formats them appropriately and pushes them back onto the appropriate
+    exchange.
+    """
+    __context_name__ = "sse"
+
+    def __init__(self, config):
+        super(SSEBackend, self).__init__(config)
+
+        self.expire_ms = int(self.config.get('fmn.pika.msg_expiration', 3600000))
+        self.amqp_host = self.config.get('fmn.pika.host', 'localhost')
+        self.amqp_port = int(self.config.get('fmn.pika.port', 5672))
+        self.amqp_parameters = pika.ConnectionParameters(self.amqp_host, self.amqp_port)
+        cc = protocol.ClientCreator(
+            reactor,
+            twisted_connection.TwistedProtocolConnection,
+            self.amqp_parameters,
+        )
+        self._deferred_connection = cc.connectTCP(self.amqp_host, self.amqp_port)
+        self._deferred_connection.addCallback(lambda conn: conn.ready)
+        self.connection = None
+        self.channel = None
+
+    @defer.inlineCallbacks
+    def handle(self, session, recipient, msg, streamline=False):
+        """
+        Handle sending a single message to one recipient.
+
+        :param session:     The SQLAlchemy database session to use.
+        :type  session:     sqlalchemy.orm.session.Session
+        :param recipient:   The recipient of the message and their settings.
+                            This controls what RabbitMQ queue the message ends
+                            up in.
+        :type recipient:    dict
+        :param msg:         The message to send to the user.
+        :type  msg:         dict
+        :param streamline:  unused
+        :param streamline:  boolean
+        """
+        short_message = self._format_message(msg, recipient)
+        user = recipient['user']
+        # TODO figure out when it's a group message
```

Comment:
That is correct, it's not there yet but we should have it at one point.

We may want to have another endpoint for groups to make things easier maybe /users/ vs /groups/?